app/forms/idv/address_form.rb

```diff
@@ -33,13 +33,21 @@ def submit(params)
 
     private
 
+    def set_ivars_with_pii(pii)
+      pii = pii.symbolize_keys
+      @address1 = pii[:address1]
+      @address2 = pii[:address2]
+      @city = pii[:city]
+      @state = pii[:state]
+      @zipcode = pii[:zipcode]
+    end
+
     def consume_params(params)
-      params.each do |key, value|
-        raise_invalid_address_parameter_error(key) unless ATTRIBUTES.include?(key.to_sym)
-        send(:"#{key}=", value)
-        if send(key) != @pii[key] && (send(key).present? || @pii[key].present?)
+      ATTRIBUTES.each do |attribute_name|
+        if send(attribute_name).to_s != params[attribute_name].to_s
           @address_edited = true
         end
+        send(:"#{attribute_name}=", params[attribute_name].to_s)
       end
```

Comment:
It looks like this is very similar to `set_ivars_with_pii` except this sets `@address_edited = true`, do you think it would be worth trying to combine? or nah

Reply:
I made an effort to combine and struggled with a good way to do it well. I am going to be working on this again in https://github.com/18F/identity-idp/pull/10385 and I'm hoping that it will be easier when we have a struct to represent an address because then we can just compare them.

Reply:
Sorry, I'll be working on this in a follow-up to #10385 since #10385 is just for reads.

Reply:
@zachmargolis: I just opened #10390 which includes these changes to the `AddressForm`: https://github.com/18F/identity-idp/pull/10390/files#diff-1b0c80e533f13f34a865f61b08ff023a3efbd21ba6bfa03aa84e0c3d0c108128

I got it out of the business of tracking address edits. My thinking is that once we stop overwriting the address in `pii_from_doc` we will get a fix for this bug: https://github.com/18F/identity-idp/pull/10385/files#diff-ee8ce27b8fbb65556458f217dd0017a6418194fd31087457ac79c219f750d337R130

Reply:
Awesome!